service/controller/chart/v1/resource/releasemigration/resource.go

```diff
@@ -53,3 +70,103 @@ func New(config Config) (*Resource, error) {
 func (r *Resource) Name() string {
 	return Name
 }
+
+func (r *Resource) findHelmV2Releases(ctx context.Context) ([]string, error) {
+	lo := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", "OWNER", "TILLER"),
+	}
+
+	// Check whether helm 2 release configMaps still exist.
+	cms, err := r.k8sClient.CoreV1().ConfigMaps(r.tillerNamespace).List(lo)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	hadReleases := map[string]bool{}
+	for _, cm := range cms.Items {
+		name, _ := cm.GetLabels()["NAME"]
+		if _, ok := hadReleases[name]; !ok {
+			hadReleases[name] = true
+		}
+	}
+
+	releases := make([]string, 0, len(hadReleases))
+	for k := range hadReleases {
+		releases = append(releases, k)
+	}
+
+	return releases, nil
+}
+
+func (r *Resource) ensureReleasesMigrated(ctx context.Context) error {
+	// 0. Found all dangling helm release v2
+	releases, err := r.findHelmV2Releases(ctx)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// 1. Install helm-2to3-migration app
+	{
+		var tarballPath string
+		{
+			tarballURL, err := appcatalog.GetLatestChart(ctx, key.DefaultCatalogStorageURL(), "helm-2to3-migration")
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			tarballPath, err = r.helmClient.PullChartTarball(ctx, tarballURL)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			defer func() {
+				fs := afero.NewOsFs()
+				err := fs.Remove(tarballPath)
+				if err != nil {
+					r.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("deletion of %#q failed", tarballPath), "stack", fmt.Sprintf("%#v", err))
+				}
+			}()
+
+			opts := helmclient.InstallOptions{
+				ReleaseName: "helm-2to3-migration",
+			}
+
+			values := map[string]interface{}{
+				"releases": releases,
+				"tiller": map[string]string{
+					"namespace": r.tillerNamespace,
+				},
+			}
+
+			err = r.helmClient.InstallReleaseFromTarball(ctx, tarballPath, r.tillerNamespace, values, opts)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
+	}
+
+	// 2. Wait until all helm v2 release are deleted
+	operation := func() error {
```

Comment:
```suggestion
	o := func() error {
```

What about using the shorter var name here?